### ISSUE TITLE
Brain LS / channels don't work

### ISSUE THREAD

**Issue by rocketinventor** [numberoverzero/brainfm#17]
Many "stations" that are shown in the CLI are actually categories, or subcategories for other stations, rather than being actual stations, themselves. This can usually be discerned by a duration value (that evaluates to) zero. For example, the "Meditate" (99), "Quick_Relax" (253), and "Nightsounds Sleep" (75) are all categories, not stations. All of them have a non-value for their "duration" property.

These channels all generate tokens, but the URL that that token corresponds to will generate a 404 error if you try to use it. ~The one exception to this is the "Focus" "station" (ID: 35) from the "Home" Group.~

These non-playable stations should be filtered out by default - they don't add much to the CLI view, but they do make it a little bit confusing.

I think that this is a similar issue to #1. Also, see #16 (about the hierarchy).

**Comment by numberoverzero** [comment]
Filtering out the stations that have zero duration value sounds like a good fix.  Are you interested in tackling this one?

I think the work involves:
* modify [`STATIONS_PATTERN`](https://github.com/numberoverzero/brainfm/blob/f17def2a9f625835beaa6259f34844ae7d757cfe/brainfm/main/cli.py#L15) to include `duration` in addition to the existing values
* post-process the [`data`](https://github.com/numberoverzero/brainfm/blob/f17def2a9f625835beaa6259f34844ae7d757cfe/brainfm/main/cli.py#L74) value to exclude 0-duration stations as you mentioned
* minor version bump

Doesn't need any readme/docs changes imo since the cli can drift from the library.

**Comment by rocketinventor** [comment]
I appreciate that you are leaving me the opportunity to make these changes (and get more involved with the code). I am interested in trying out the steps that you mentioned. They seem easy enough to do - trivial even.

The only problem is that I am unsure of the steps needed to cleanly switch to/from a development version of this package (as opposed to the PyPi version).

**Comment by numberoverzero** [comment]
1. clone to a local directory:
   `git clone https://github.com/numberoverzero/brainfm.git`
2. **(optional)** create a new virtual environment to develop in.  you can use your existing env as well.
   ```
   cd brainfm
   python3 -m venv --copies .venv-develop
   source .venv-develop
   ```
3. install the git repo you just cloned as an editable package
   `pip install -e .` (from the repo directory you cloned to)
   The `-e` will tell pip to install this as an editable package, creating symlinks so you don't need to re-install each change.

Let me know if you have trouble with any of those steps.

**Comment by rocketinventor** [comment]
Thank you for those instructions. I searched around for a while, but I wasn't able to find those instructions anywhere else. The `source` command didn't work for me (on Windows), so I just skipped the entire step two.

To go back to the registry version I just run `pip install brainfm`?

**Comment by rocketinventor** [comment]
With the help of your instructions, I was able to get the filtering working without a problem.

I decided, also, to add a "length" column to the table. I think that it is a useful addition:

```
+Available Stations------------+--------------------------------------------+--------+
| id  | name                   | string_id                                  | length |
+-----+------------------------+--------------------------------------------+--------+
| 32  | Quick Relax            | relax.justrelax15min                       | 15     |
| 34  | Relaxed Focus          | explore.relaxed                            | 30     |
| 35  | Focus                  | focus.3                                    | 30     |
```
Perhaps, in a later release, we could convert these numbers into hrs/minute amounts. The `README.rst` file will need to be updated.

I am ready to release... Does the version number need to get updated anywhere else besides for line `6` in `__init__.py`? Does this count as V2.5? Or 2.1.1? Or 3.0? What exactly is your naming convention?

Any feedback?

**Comment by rocketinventor** [comment]
Should there be an `ls -a` command to show _all_ the stations? Should I make a new issue for this?

**Comment by numberoverzero** [comment]
Nice work, length is a great addition!  Since it does change the table format, this is a breaking change and should be bumped to 3.0 (excluding unplayable is also a breaking change, I guess)

I've been using breaking change/new feature/non-breaking bug fix as the major/minor/patch convention.  Since both of these will modify an output that users may have been parsing, and they couldn't reasonably anticipate these changes, that makes them both breaking.

---

Since the `-a` change should be small and you're already working on that part of the code, feel free to include it in the same PR or split it out how you'd like.  Please mention the flag in the readme as well.

Really excited to see a pull request, thanks for doing this work! ❤️

**Comment by rocketinventor** [comment]
@numberoverzero Feel free to close this issue as soon as the pull-request is approved. Thank you for walking me through the changes that needed to be made!

**Comment by numberoverzero** [comment]
Merged #18, closing.  Nice work :)
